platform/linux-generic/odp_packet_io.c

```diff
@@ -283,53 +230,83 @@ static const char *strip_pktio_type(const char *name, char *type_out)
 	return name;
 }
 
+static void init_out_queues(pktio_entry_t *entry)
+{
+	int i;
+
+	for (i = 0; i < PKTIO_MAX_QUEUES; i++) {
+		entry->s.out_queue[i].queue  = ODP_QUEUE_INVALID;
+		entry->s.out_queue[i].pktout = PKTOUT_INVALID;
+	}
+}
+
+static void init_pktio_entry(pktio_entry_t *entry)
+{
+	int i;
+
+	/* Clear all flags */
+	entry->s.enabled.all_flags = 0;
+
+	odp_atomic_init_u64(&entry->s.stats_extra.in_discards, 0);
+	odp_atomic_init_u64(&entry->s.stats_extra.out_discards, 0);
+	odp_atomic_init_u64(&entry->s.tx_ts, 0);
+
+	for (i = 0; i < PKTIO_MAX_QUEUES; i++) {
+		entry->s.in_queue[i].queue = ODP_QUEUE_INVALID;
+		entry->s.in_queue[i].pktin = PKTIN_INVALID;
+	}
+
+	init_out_queues(entry);
```

Comment:
Both queue inits done in a function or neither?

Reply:
Could not remove it since TM (_odp_pktio_pktout_tm_config()) calls this function as well.